src/Symfony/Component/OptionsResolver/Tests/OptionsResolverTest.php

```diff
@@ -474,6 +474,168 @@ public function testClearedOptionsAreNotDefined()
+                ;
+            },
+            array('foo' => 'baz'),
+            array(
+                'type' => E_USER_DEPRECATED,
+                'message' => 'The option "foo" is deprecated, use "bar" option instead.',
+            ),
+        );
+
+        yield 'It ignores deprecation for missing option without default' => array(
+            function (OptionsResolver $resolver) {
+                $resolver
+                    ->setDefined(array('foo', 'bar'))
+                    ->setDeprecated('foo')
+                ;
+            },
+            array('bar' => 'baz'),
+            null,
+        );
+
+        yield 'It deprecates a missing option with default value' => array(
+            function (OptionsResolver $resolver) {
+                $resolver
+                    ->setDefaults(array('foo' => null, 'bar' => null))
+                    ->setDeprecated('foo')
+                ;
+            },
+            array('bar' => 'baz'),
+            array(
+                'type' => E_USER_DEPRECATED,
+                'message' => 'The option "foo" is deprecated.',
+            ),
+        );
+
+        yield 'It deprecates allowed type and value' => array(
+            function (OptionsResolver $resolver) {
+                $resolver
+                    ->setDefault('foo', null)
+                    ->setAllowedTypes('foo', array('null', 'string', Bar::class))
+                    ->setDeprecated('foo', function ($value) {
+                        if ($value instanceof Bar) {
+                            return sprintf('Passing an instance of "%s" to option "foo" is deprecated, pass its FQCN instead.', Bar::class);
+                        }
+                    })
+                ;
+            },
+            array('foo' => new Bar()),
+            array(
+                'type' => E_USER_DEPRECATED,
+                'message' => 'Passing an instance of "Symfony\Component\OptionsResolver\Tests\Bar" to option "foo" is deprecated, pass its FQCN instead.',
+            ),
+        );
+
+        yield 'It ignores deprecation if closure does not return a string' => array(
+            function (OptionsResolver $resolver) {
+                $resolver
+                    ->setDefault('foo', null)
+                    ->setAllowedTypes('foo', array('null', 'string', Bar::class))
+                    ->setDeprecated('foo', function ($value) {
+                        if ($value instanceof Bar) {
+                            return sprintf('Passing an instance of "%s" to option "foo" is deprecated, pass its FQCN instead.', Bar::class);
+                        }
+                    })
+                ;
+            },
+            array('foo' => Bar::class),
+            null,
+        );
+    }
+
     ////////////////////////////////////////////////////////////////////////////
     // setAllowedTypes()
     ////////////////////////////////////////////////////////////////////////////
```

Comment:
I see than you did the same as here, I'm going to remove such comments as they are not useful.

Reply:
Yes, just for consistency. I had never seen these types of comments in other tests.